### ISSUE TITLE
Path errors with OSX 10.10.3

### ISSUE THREAD

**Issue by dkoontz** [atom-haskell/haskell-ghc-mod#25]
On startup I'm getting the error:

`Failed to spawn command /Users/dkoontz/Library/Haskell/bin/ghc-mod. Make sure /Users/dkoontz/Library/Haskell/bin/ghc-mod is installed and on your PATH`

I have configured path in the settings for haskell-ghc-mod
```
"haskell-ghc-mod":
    additionalPathDirectories: [
      "/usr/bin/ghc"
    ]
    ghcModPath: "/Users/dkoontz/Library/Haskell/bin/ghc-mod"
    ghcModiPath: "/Users/dkoontz/Library/Haskell/bin/ghc-modi"
```
If I open my terminal and run `/Users/dkoontz/Library/Haskell/bin/ghc-mod version` I can confirm that I get `ghc-mod version 5.2.1.2 compiled by GHC 7.8.3`

When I hover over an identifier in a .hs file I get a second error 

`'/Users/dkoontz/Library/Haskell/bin/ghc-modi' could not be spawned. Is it installed and on your path? If so please open an issue on the package spawning the process.`

I'm fairly stumped by what could be happening here. I've tried opening Atom from the command line, I've tried putting an explicit path in init.coffee as some other posts on this topic have suggested
```
process.env.PATH = ["/usr/bin",
                    "/usr/local/bin",
                    "/Users/dkoontz/Library/Haskell/bin",
                    process.env.PATH
                    ].join(":")
```
but that seems to have no effect.

**Comment by lierdakil** [comment]
Is `/Users/dkoontz/Library/Haskell/bin/ghc-modi` available though? I gather `/Users/dkoontz/Library/Haskell/bin/ghc-mod` is, but is `ghc-modi` there? It looks like it's not.

If it's not, you have a couple options. First would be to disable `Enable Ghc-Modi` checkbox, but that would make type/info search slow.
Second is to create a bash/zsh script `ghc-modi` with following contents (bash version):

```bash
#!/bin/bash

/Users/dkoontz/Library/Haskell/bin/ghc-mod legacy-interactive "$@"
```
`chmod +x` it, and use it instead ghc-modi executable.

That said, `ghc-modi` should be included in ghc-mod bundle, so I'm not sure why could it be missing.

**Comment by lierdakil** [comment]
Oh. I somehow missed first paragraph of your message. Well, then I'm as stumped as you are.

One thing you *might* try is to put `/Users/dkoontz/Library/Haskell/bin` to `additionalPathDirectories`, and leave `ghcModPath` and `ghcModiPath` at default values. I'm not convinced that would work though.

**Comment by dkoontz** [comment]
So if I remove the explicit path to ghc-mod / ghc-modi I get an error saying "can't find ghc-mod". When I put in the explicit path I get the `Uncaught BufferedProcessError: Failed to spawn command`. So it seems like perhaps Atom is finding the executable but is not able to invoke it. Is there any sort of debugging I could do on the binary invocation?

**Comment by lierdakil** [comment]
This might be a permissions problem, maybe?

Check developer console for any messages that might give some insight. If there aren't any, might try to run a couple of commands from it to see what goes wrong.

First thing you should try running from dev console is
```javascript
require('child_process').execFileSync('GHC-MOD',['version'])+""
```
replace `GHC-MOD` with full path to ghc-mod executable.

If that works allright, check with Atom's setting next:
```js
require('child_process').execFileSync(atom.config.get('haskell-ghc-mod.ghcModPath'),['version'])+""
```

If nothing provides insight, we'll have to go deeper.

**Comment by dkoontz** [comment]
So I tried the commands you suggested.

> require('child_process').execFileSync('GHC-MOD',['version'])+""

"GHC-MOD version 5.2.1.2 compiled by GHC 7.8.3"

>
require('child_process').execFileSync('/Users/dkoontz/Library/Haskell/bin/ghc-mod',['version'])+""

"ghc-mod version 5.2.1.2 compiled by GHC 7.8.3"

>
require('child_process').execFileSync(atom.config.get('haskell-ghc-mod.ghcModPath'),['version'])+""

"ghc-mod version 5.2.1.2 compiled by GHC 7.8.3"

So all of those commands seem to be working, even leaving 'GHC-MOD' as the
command without a fully qualified path. Just to sanity check things I tried
altering the command to run to be something else

> require('child_process').execFileSync('foobarbaz',['version'])+""

and that correct produces an error: Uncaught TypeError: Cannot read
property 'length' of null

so that would indicate the ghc-mod executable is runnable by Atom, at least
through the dev console.

**Comment by lierdakil** [comment]
A quick sanity check, what Atom version do you use?

**Comment by dkoontz** [comment]
I was testing on 0.207.0 but I see there's an update to 0.209.0. After
updating the loading errors are gone! Strange since I just downloaded Atom
a few days ago, I guess I missed the update somehow.

**Comment by lierdakil** [comment]
Currently, Atom updates very frequently. They're working hard on releasing 1.0 by the end of month, so release cycle is very short right now.

If everything works after update, I'll chalk this up to Atom bugs. Please confirm that I can close this.

**Comment by dkoontz** [comment]
Yes everything seems ok now. For reference, I removed the extra entry in
the additionPathDirectories field and everything still works, so just
specifying the path to ghc-mod and ghc-modi is sufficient.

**Comment by diejmon** [comment]
I have Atom version 1.0.0 and got the same error as dkoontz. No matter what I do it says: 
ghc-mod debug: ghc-mod ended with code 0
Failed to spawn command ghc-mod. Make sure ghc-mod is installed and on your PATH

require('child_process').execFileSync('GHC-MOD',['version'])+""
"GHC-MOD version 5.2.1.2 compiled by GHC 7.8.3

 require('child_process').execFileSync(atom.config.get('haskell-ghc-mod.ghcModPath'),['version'])+""
"ghc-mod version 5.2.1.2 compiled by GHC 7.8.3

**Comment by lierdakil** [comment]
@diejmon, if you're comfortable with communicating in Russian, you might consider contacting me directly via e-mail (check GitHub profile) or jabber (same as e-mail). I'd like to try to get to bottom of this, but at the moment I have no idea what could be wrong.

In any case, first question I'd like to ask is what OS you're running Atom on.

**Comment by lierdakil** [comment]
Okay, I've released 0.7.11, which should at least give more verbose error
messages. Check dev console for errors/warnings. Maybe this will provide
some insight.

2015-06-28 14:09 GMT+03:00 Alexander Belyavskiy <notifications@github.com>:

**Comment by diejmon** [comment]
0.7.12 works for me. Thanks for fix.
